inference-engine/thirdparty/clDNN/tests/test_cases/cl_mem_input_test.cpp

```diff
@@ -304,3 +304,73 @@ TEST(cl_mem_check, check_input) {
     }
     checkStatus(clReleaseMemObject(img), "clReleaseMemObject");
 }
+
+TEST(cl_mem_check, check_write_access_type) {
+    auto ocl_instance = std::make_shared<OpenCL>();
+
+    device_query query(engine_types::ocl, runtime_types::ocl, static_cast<void*>(ocl_instance->_context.get()));
+    auto device = query.get_available_devices().begin()->second;
+
+    auto engine = std::dynamic_pointer_cast<ocl::ocl_engine>(engine::create(engine_types::ocl, runtime_types::ocl, device));
+    ocl::ocl_stream stream(*engine);
+
+    size_t values_count = 100;
+    size_t values_bytes_count = values_count * sizeof(float);
+    std::vector<float> src_buffer(values_count);
+    std::iota(src_buffer.begin(), src_buffer.end(), 0.0f);
+
+    cldnn::layout linear_layout = cldnn::layout(cldnn::data_types::f32, cldnn::format::bfyx, cldnn::tensor(1, 1, int32_t(values_count), 1));
+    auto cldnn_mem_src = engine->allocate_memory(linear_layout, cldnn::allocation_type::cl_mem);
+    {
+        cldnn::mem_lock<float, cldnn::mem_lock_type::write> lock(cldnn_mem_src, stream);
+        std::copy(src_buffer.begin(), src_buffer.end(), lock.data());
+    }
+
+    std::vector<float> dst_buffer(values_count);
+    {
+        cldnn::mem_lock<float, cldnn::mem_lock_type::read> lock(cldnn_mem_src, stream);
+        std::memcpy(dst_buffer.data(), lock.data(), values_bytes_count);
+    }
+
+    bool are_equal = std::equal(src_buffer.begin(), src_buffer.begin() + values_count, dst_buffer.begin());
+    ASSERT_EQ(true, are_equal);
+}
+
+TEST(cl_mem_check, check_read_access_type) {
+    auto ocl_instance = std::make_shared<OpenCL>();
+
+    device_query query(engine_types::ocl, runtime_types::ocl, static_cast<void*>(ocl_instance->_context.get()));
+    auto device = query.get_available_devices().begin()->second;
+    if (device->get_info().dev_type == device_type::integrated_gpu) {
+        GTEST_SKIP();
+    }
+
+    auto engine = std::dynamic_pointer_cast<ocl::ocl_engine>(engine::create(engine_types::ocl, runtime_types::ocl, device));
+    ocl::ocl_stream stream(*engine);
```

Comment:
Why you didn't use `get_test_engine()` and `get_test_stream()` functions? The same device (first available) will be used as default for engine configuration in case of `get_test_engine()` call
```
auto& engine = get_test_engine();
auto& stream = get_test_stream();
```

Reply:
Because it uses an integrated GPU even if a discrete is available

Reply:
But as for now you use exactly the same approach: `auto device = query.get_available_devices().begin()->second;`. Isn't it?